### DISCUSSION TITLE
all zero in statistical info

### DISCUSSION THREAD

**Discussion by d1115choi** [lldpd/lldpd#571] [Q&A]
Hello LLDP expters,

I ported 1.0.15 into ARM based system. Generally it works fine.
There are cases that we need to restart my system several times. Whenever the system restart, lldpd also restarts.
I am facing an stragne situation that "lldpcli show statistics" show all zero.
It is very difficult to use the "-d" option in my test environment.
In this situation, 
-ping works fine.
-tcpdump shows that there is LLDPDPU incoming very 30second.

Is there any clue causing the issue?

**Comment by d1115choi** [comment]
Is there any possiblity that c_pause=1 is set after executing "lldpcli resume" so that lldpd is in virtually pause state ,not sending nor receiving(recevie but ignore the LLDPDU)?

**Reply by vincentbernat** [reply]
You can run `lldpcli resume` in this case. This can happen if lldpd does not know the path to lldpcli.

**Reply by d1115choi** [reply]
Hi Vincentbernat,

When the issue happened, running "lldpcli resume" maked it work. This issue happens in certain environment. does not in another environment.( I am still digging the root cause.)

After tracing the issue by myself, here is what I found so far. The sequence of invocatations:
1. A child process(llcpcki is invoked) with "resume" is triggered to start.
2. In the parent process, c_pause is set 1.

If the parent process sets c_pause=1 before running the child process, I think the mechnism works as expected because the "resume" codes triggers to reset c_pauses=0.  

Here is my argument.
What happens if the parent process c_pause is set if the the child process finishes the "resume" process?

So, if we switch the sequence 1 and 2, I assume that the issue does not happen in any case.

**Reply by vincentbernat** [reply]
Parent set `c_paused` to 1, then it invokes lldpcli. What can happen is lldpcli can fail. There is no way lldpcli can run before parent set `c_paused` to 1.

**Reply by d1115choi** [reply]
Thank you so much for your info.
